common/src/main/java/org/opensearch/ml/common/transport/remote/MLRemoteModelInput.java

```diff
@@ -0,0 +1,177 @@
+package org.opensearch.ml.common.transport.remote;
+
+import lombok.Builder;
+import lombok.Data;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.io.stream.Writeable;
+import org.opensearch.common.xcontent.ToXContentObject;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.common.xcontent.XContentParser;
+import org.opensearch.ml.common.model.TextEmbeddingModelConfig;
+import org.opensearch.ml.common.transport.upload.MLUploadInput;
+
+import java.io.IOException;
+
+import static org.opensearch.common.xcontent.XContentParserUtils.ensureExpectedToken;
+
+@Data
+public class MLRemoteModelInput implements ToXContentObject, Writeable {
+
+    public static final String REMOTE_MODEL_NAME_FIELD = "remote_model_name";
+    public static final String DESCRIPTION_FIELD = "description";
+    public static final String VERSION_FIELD = "version";
+    public static final String ENDPOINT = "endpoint";
+    public static final String ACCESS_KEY_ID = "access_key_id";
+    public static final String SECRET_ACCESS_KEY = "secret_access_key";
+
+    private String remoteModelName;
+    private String description;
+    private String version;
+    private String endpoint;
```

Comment:
from line 31-33,these 3 variable in input class,If we used those in input parameters, whatever in rest or tcp transfer case,is it will trigger security problem? I remember that things like id&key can't be passed in and out as plain text in the parameters, and it's not correct if their values are written in the code(will it be written in yml file?)